GitUI/CommandsDialogs/FormFileHistory.cs

```diff
@@ -233,64 +246,33 @@ private void LoadFileHistory()
 
                 FileName = fileName;
 
-                var res = (revision: (string)null, path: $" \"{fileName}\"");
+                var res = (revision: (ArgumentBuilder)new ArgumentBuilder(), path: fileName);
 
-                if (AppSettings.FollowRenamesInFileHistory && !Directory.Exists(fullFilePath))
+                if (AppSettings.FollowRenamesInFileHistory)
                 {
-                    // git log --follow is not working as expected (see  http://kerneltrap.org/mailarchive/git/2009/1/30/4856404/thread)
```

Comment:
You're effectively removing this workaround, right? Are you sure it's no longer required? (The linked Git mailing list post can be found [here](https://www.spinics.net/lists/git/msg92132.html) as the original URL seems to be broken. Thread topic for googling: "Git log can not show history before rename".)

Reply:
Why use a workaround when the problem the workaround solves goes away?

Reply:
The workaround was added in https://github.com/gitextensions/gitextensions/commit/e1b029d207624c26c1ed0143b772afc87f10f683 with this message:

> Directly using --follow can hide some commits relevant (even if the file was never renamed) - this approach might show more commits than needed.

It was changed later to resolve #2452. I can't really tell if these issues have gone away :)

Reply:
It will. It is because of the parsing that we were only getting first file name.